# HG changeset patch
# User Pantxo Diribarne <pantxo.diribarne@gmail.com>
# Date 1551637047 -3600
#      Sun Mar 03 19:17:27 2019 +0100
# Node ID f9b10a8a7c93cf3a356171d11ded6261d9546799
# Parent  0a383bddfb404387963bd25d086cebda3b4e30be
Allow changing the position of a maximized/fullscreen Qt figure.

* Figure.cc (Figure::set_geometry): Change the figure window state to
  Qt::WindowNoState before attempting to resize it.

diff --git a/libgui/graphics/Figure.cc b/libgui/graphics/Figure.cc
--- a/libgui/graphics/Figure.cc
+++ b/libgui/graphics/Figure.cc
@@ -302,16 +302,21 @@ namespace QtHandles
     QMainWindow *win = qWidget<QMainWindow> ();
 
     if (! m_resizable)
       {
         win->setSizePolicy (QSizePolicy::Preferred, QSizePolicy::Preferred);
         win->setFixedSize (QSize( QWIDGETSIZE_MAX, QWIDGETSIZE_MAX));
       }
 
+    // Unlock window if it is maximized or full-screen
+    int state = win->windowState ();
+    if (state == Qt::WindowFullScreen || state == Qt::WindowMaximized)
+      win->setWindowState (Qt::WindowNoState);
+
     win->setGeometry (r);
 
     if (! m_resizable)
       {
         win->setSizePolicy (QSizePolicy::Fixed, QSizePolicy::Fixed);
         win->setFixedSize(win->size ());
       }
   }
